### PR TITLE
Add PDB on linkerd-viz Helm chart

### PR DESCRIPTION
**Pull request by mcharriere**
To avoid disruption during cluster nodes rolling or scaling, it's desired to complement the maxUnavailable from the rolling strategy with a PodDisruptionBudget.

This commit adds the respective PDB objects following the core components implementation to the `tap`, `tap-injector`, `web` and `metrics-api` deployments. It can be enabled with the enablePodDisruptionBudget helm chart value and are installed only if each of those components have more than 1 replica.

Fixes https://github.com/linkerd/linkerd2/issues/11248

Previous PR was close due inactivity (https://github.com/linkerd/linkerd2/pull/11249), sorry about that.

> I'm wondering if the PodDisruptionBudgets should apply to the other deployments in linkerd-viz as well, namely web, prometheus, and metrics-api. What do you think?

As requested there, I added in this new PR PDBs for `web` and `metrics-api`. `prometheus` has only fixed 1 replica; so it doesn't really make sense to add a PDB for it.

### CODE REVIEW

**Review by adleong — APPROVED** [review]
nice!

File: viz/charts/linkerd-viz/values.yaml [hunk 1]
```diff
@@ -50,6 +50,9 @@ tolerations: &default_tolerations
 # Enable this only when you have multiple replicas of components.
 enablePodAntiAffinity: false
 
+# -- enables the creation of pod disruption budgets for tap and tap-injector components
```
Comment (alpeb) [inline viz/charts/linkerd-viz/values.yaml:53]: Can you please add `web` to the list, and re-run `bin/helm-docs`?

Comment (mcharriere) [reply viz/charts/linkerd-viz/values.yaml]: done. also added metrics-api.

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Pull request merged**